restapi.go

```diff
@@ -2328,3 +2350,141 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
 	err = unmarshal(body, &mf)
 	return
 }
+
+// ------------------------------------------------------------------------------------------------
+// Functions specific to application (slash) commands
+// ------------------------------------------------------------------------------------------------
+
+// ApplicationCommandCreate creates an global application command. If guild is specified - creates guild specific application command.
+func (s *Session) ApplicationCommandCreate(appID string, cmd *ApplicationCommand, guildID string) (rcmd *ApplicationCommand, err error) {
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+	endpoint := EndpointApplicationGlobalCommands(s.State.User.ID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommands(s.State.User.ID, guildID)
+	}
+	body, err := s.RequestWithBucketID("POST", endpoint, *cmd, endpoint)
+	if err == nil {
+		err = unmarshal(body, &rcmd)
+	}
+
+	return
+}
+
+// ApplicationCommandDelete deletes global application command by id. If guild is specified - deletes guild specific application command.
+func (s *Session) ApplicationCommandDelete(appID, cmdID, guildID string) (err error) {
+	if appID == "" {
```

Comment:
I'm not convinced this is the best idea. If we are going through with it, at least document it in the comment.

Reply:
I agree—this is a confusing API surface, and might confuse people using the method in a webhook-only capacity to think that passing `""` would resolve their application, when it wouldn't.

Reply:
Alright, I will change docs.

Reply:
The best reaction is return error, if user don't send some data. U don't need think out for the user

Reply:
And many parts in code seems like `a := myfunc("", "")` - really bad code style

Reply:
No, it isn't, at least there's only one parameter which might be `""`

Reply:
I would also prefer if this didnt try to second guess the user, but this isnt _as_ big an issue for me if its documented. That being said, often times users dont even read the docs :roll_eyes:

Reply:
Oh, yeah, I forget to add the note about this here. My bad.